### PR TITLE
chore: dependency updates

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

- Needs  https://github.com/strapi/strapi/pull/14017 to be merged 1st
- Does a bunch of dependencies changes and cleanups to fix warnings and errors

### Why is it needed?

- Very obvious :D 

### How to test it?

you can try the experimental release of these changes with `npx create-strapi-app@experimental my-app`

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/core/admin/package.json [hunk 1]
```diff
@@ -114,6 +113,7 @@
     "react-fast-compare": "^3.2.0",
     "react-helmet": "^6.1.0",
     "react-intl": "5.20.2",
+    "react-is": "^17.0.2",
```
Comment (gu-stav) [inline packages/core/admin/package.json:116]: I'm still wondering where this comes from. We don't use the package explictly anywhere ...

File: packages/core/admin/package.json [hunk 2]
```diff
@@ -128,6 +128,7 @@
     "sift": "13.5.4",
     "style-loader": "3.3.1",
     "styled-components": "5.3.3",
+    "typescript": "4.6.2",
```
Comment (gu-stav) [inline packages/core/admin/package.json:131]: Do we have a hard dependency on typescript?

Comment (alexandrebodin) [reply packages/core/admin/package.json:131]: the webpack ts-checker plugin has one apparently

File: package.json [hunk 3]
```diff
@@ -107,7 +104,10 @@
     "plop": "2.7.6",
     "prettier": "2.7.1",
     "qs": "6.11.0",
-    "react-test-renderer": "17.0.2",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-test-renderer": "^17.0.2",
+    "redux": "^4.0.1",
```
[the remark this answers is not in the document]
Comment (gu-stav) [reply package.json:110]: Ok then. Would you like on the follow-up or should someone else? I think we'd really only need to move all `@testing-library/` packages into `packages/*`. I tried that in https://github.com/strapi/strapi/pull/14030 but combined it with package updates (which leads to 3 failed tests :/).

File: packages/core/admin/admin/src/components/LanguageProvider/tests/index.test.js [hunk 4]
```diff
@@ -56,7 +56,7 @@ describe('LanguageProvider', () => {
 
     userEvent.click(screen.getByText('CHANGE'));
 
-    expect(screen.getByText('Français')).toBeInTheDocument();
+    await waitFor(() => expect(screen.getByText('Français')).toBeInTheDocument());
```
[the remark this answers is not in the document]
Comment (gu-stav) [reply packages/core/admin/admin/src/components/LanguageProvider/tests/index.test.js:59]: I'm wondering why this effect needs to be async. Would:

```js
act(() => {
  userEvent.click(screen.getByText('CHANGE'));
})

expect(screen.getByText('Français')).toBeInTheDocument();
```

work?

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexandrebodin) [reply package.json:110]: Applied in this PR

**Comment by gu-stav** [comment]
@alexandrebodin Looks like we need to add back the `react-test-renderer` dependency 🤔 ...

**Comment by alexandrebodin** [comment]
> @alexandrebodin Looks like we need to add back the `react-test-renderer` dependency 🤔 ...

Yes it's supposed to be optional but react-hooks seem to require it

File: packages/core/admin/admin/src/components/LanguageProvider/tests/index.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (gu-stav) [reply packages/core/admin/admin/src/components/LanguageProvider/tests/index.test.js]: Looks like as if this is going to work. 👍🏼

**Review by gu-stav — APPROVED** [review]
LGTM!

**Comment by alexandrebodin** [comment]
I'll wait for the eslint PR to be approved to merge this one 👍

**Pull request merged**